File: NextAppSzakemberKereso/src/utils/pricing.js
export const PricingType = {
  Fixed: 0,
  FixedAndUnitBased: 1,
}

export function formatPricing(pricing) {
  if (!pricing) return null;
  switch (pricing.pricingType) {
    case PricingType.Fixed:
      return `${pricing.fixedPrice.toLocaleString("hu-HU")} Ft`;
    case PricingType.FixedAndUnitBased:
      if (pricing.fixedPrice > 0) {
        return `${pricing.fixedPrice.toLocaleString("hu-HU")} Ft + ${pricing.unitPrice?.toLocaleString("hu-HU")} Ft / ${pricing.unitName}`;
      }
      return `${pricing.unitPrice?.toLocaleString("hu-HU")} Ft / ${pricing.unitName}`;
    default:
      return null;
  }
}

File: BackendSzakemberKereso/SzakemberKereso/Models/Pricing.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace SzakemberKereso.Models
{
    public enum PricingType
    {
        Fixed = 0,
        FixedAndUnitBased = 1
    }

    [Table("pricing")]
    public class Pricing
    {
        [Key]
        [Column("pricing_id")]
        public int Id { get; set; }

        [Column("pricing_type")]
        public PricingType PricingType { get; set; }

        [Column("fixed_price", TypeName = "decimal(18,2)")]
        public decimal FixedPrice { get; set; }

        [Column("unit_price", TypeName = "decimal(18,2)")]
        public decimal? UnitPrice { get; set; }

        [Column("unit_name")]
        [StringLength(64)]
        public string? UnitName { get; set; }
    }
}
